### PR TITLE
Added support for alipay.trade.app.pay method

### PR DESCRIPTION
**Pull request by FX-HAO**

### CODE REVIEW

**Comment by FX-HAO** [comment]
https://github.com/chloerei/alipay/issues/63

**Comment by chloerei** [comment]
Is it independent from "mobile" API? If yes I think it can create a new
module named "APP".
Fuxin Hao <notifications@github.com>于2016年12月22日 周四12:19写道：

File: lib/alipay/sign.rb [hunk 1]
```diff
@@ -47,6 +56,26 @@ def self.verify?(params, options = {})
...
+      sign_type = params.delete('sign_type')
+      sign = params.delete('sign')
+      string = params_to_string(params)
+
+      case sign_type
+      when 'MD5'
+        key = options[:key] || Alipay.key
+        MD5.verify?(key, string, sign)
+      when 'RSA'
+        RSA.verify?(ALIPAY_RSA_APP_PAY_PUBLIC_KEY, string, sign)
```
Comment (rei-huang) [inline lib/alipay/sign.rb]: It should be `Alipay::Sign:: RSA`

File: lib/alipay/sign.rb [hunk 2]
```diff
@@ -47,6 +56,26 @@ def self.verify?(params, options = {})
...
 
+    def self.app_pay_verify?(params, options = {})
+      params = Utils.stringify_keys(params)
+
+      sign_type = params.delete('sign_type')
+      sign = params.delete('sign')
+      string = params_to_string(params)
+
+      case sign_type
+      when 'MD5'
```
Comment (rei-huang) [inline lib/alipay/sign.rb]: Is it only support RSA sign_type, then MD5 and DSA should be delete.

File: lib/alipay/sign.rb (hunk 1 above)
Comment (FX-HAO) [inline lib/alipay/sign.rb]: It's ok to be RSA.

Comment (rei-huang) [inline lib/alipay/sign.rb]: Oh, I miss the filename, this sign logic should move to `alipay/app/sign.rb`, like `alipay/wap/sign.rb`.

And service methods need a unit test.

**Comment by FX-HAO** [comment]
Updated it.

**Pull request merged**